### PR TITLE
chore: add controllable logging support

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

add a logging backend and interface with options to filter the log statements

## Commentary (optional)

part of https://github.com/determined-ai/saas/pull/290
https://determined-ai.slack.com/archives/CKQ3EV0G6/p1660592735267419

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/shared/utils/routes.ts [hunk 1]
```diff
@@ -3,6 +3,9 @@ import React from 'react';
 import history from '../routes/history';
 
 import { clone } from './data';
+import rootLogger from './Logger';
+
+export const logger = rootLogger.extend('utils', 'routes');
```
Comment (hamidzr) [inline webui/react/src/shared/utils/routes.ts:8]: avoid exporting?

File: webui/react/src/shared/utils/error.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
-import Logger from 'shared/utils/Logger';
+import rootLogger from 'shared/utils/Logger';
 
 import { isString } from './data';
 import { LoggerInterface } from './Logger';
 
 export const ERROR_NAMESPACE = 'EH';
 export const DEFAULT_ERROR_MESSAGE = 'Unknown error encountered.';
-export const DEFAULT_LOGGER = new Logger(ERROR_NAMESPACE);
+export const DEFAULT_LOGGER = rootLogger.extend(ERROR_NAMESPACE);
```
Comment (hamidzr) [inline webui/react/src/shared/utils/error.ts:8]: why did we export this?

**Review by johnkim-det — APPROVED** [review]
not sure if i have context to totally understand all the changes but not seeing any problems.

**Comment by hamidzr** [comment]
> not sure if i have context to totally understand all the changes but not seeing any problems.

I'd be happy to go through it tomorrow to make sure it all makes sense before we start using it everywhere

**Comment by hamidzr** [comment]
downstream test failures are expected here as we're adding a new package.

**Pull request merged**